src/components/header/bookmark.tsx

```diff
@@ -1,11 +1,100 @@
 import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import Modal from 'react-modal';
+import {connect} from 'react-redux';
+import {BookmarkAction} from '../../actions/bookmark';
+import {ActionHandler, createDispatchHandler} from '../../actions/redux-action';
+import {State} from '../../models';
+import {Bookmark} from '../../models/bookmark';
+import {PlotObject} from '../../models/plot';
+import {Plot} from '../plot';
+import * as styles from './bookmark.scss';
+
+
+
+export interface BookmarkProps extends ActionHandler<BookmarkAction> {
+  bookmark: Bookmark;
+}
+
+export class BookmarkBase extends React.PureComponent<BookmarkProps, any> {
+  constructor(props: BookmarkProps) {
+    super(props);
+
+    this.state = {modalIsOpen: false};
+    this.openModal = this.openModal.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
 
-export class Bookmark extends React.PureComponent<{}, {}> {
   public render() {
     return (
-      <button>
-        <i className='fa fa-bookmark' /> Bookmarks (0)
-      </button>
+      <div>
+        <button onClick={this.openModal}>
+          <i className='fa fa-bookmark' /> Bookmarks ({this.props.bookmark.numBookmarks})
+        </button>
+
+        <Modal
+          isOpen={this.state.modalIsOpen}
+          onRequestClose={this.closeModal}
+          contentLabel="Bookmark Selector"
+          styleName="modal"
+        >
+          {this.renderBookmarks(this.props.bookmark)}
+        </Modal>
+      </div>
+    );
+  }
+
+  private openModal() {
+    this.setState({modalIsOpen: true});
+  }
+
+  private closeModal() {
+    this.setState({modalIsOpen: false});
+  }
+
+  private renderBookmarks(bookmark: Bookmark) {
+    const bookmarks = bookmark.dict;
+    const plots: PlotObject[] = [];
```

Comment:
Seeing this line, I'd say you should include `bookmark.list` in this PR and just read it from there so we can get rid of dictionary reading here.